enterprise-suite/tests/smokecommon

```diff
@@ -32,8 +32,37 @@ busy_wait() {
   done
 }
 
-TEST_SUCCESSES=0
-TEST_FAILURES=0
+timeout() {
+    # default timeout is 90s.  If $1 is a number, it will be used as timeout value
+    timeout=90
+    re='^[0-9]+$'
+    if [[ "$1" =~ $re ]] ; then
+        timeout=$1
+        shift
+    fi
+    timeout_function $timeout "$@"
+}
+
+
+# Setup fds for writing/reading success/failure counts.  Allows things to work across subshells
+SUCCESSES=$(mktemp -t pipe.XXXXX)
+exec 3>$SUCCESSES
+exec 4<$SUCCESSES
+# unlink so we don't have to clean up later
+rm $SUCCESSES
+
+FAILURES=$(mktemp -t pipe.XXXXX)
+exec 6>$FAILURES
+exec 7<$FAILURES
+# unlink so we don't have to clean up later
+rm $FAILURES
+
+TIMEOUTS=$(mktemp -t pipe.XXXXX)
+exec 8>$TIMEOUTS
+exec 9<$TIMEOUTS
+# unlink so we don't have to clean up later
+rm $TIMEOUTS
+
```

Comment:
Would this be simpler if we used `mkfifo` instead? Then we can write/read from the named pipes rather than anonymous file descriptors, which I find pretty hard to follow.

Reply:
I originally tried with mkfifo but it just wouldn't work.  I would close it for writing but that would close it for reading at the same time and it would disappear.  I tried various stunts but all to no avail.  It just worked as soon as I moved to an actual file.

I also tried to use variables for the file descriptors but that got even uglier fast.  (cf. https://stackoverflow.com/questions/8295908/how-to-use-a-variable-to-indicate-a-file-descriptor-in-bash)  And I skipped 5 cuz some page suggested that was a good idea...

So it is what it is.